### PR TITLE
PRoject 5: Guess Who?

### PR DESCRIPTION
**Pull request by rubmaigut**

### CODE REVIEW

File: code/script.js [hunk 1]
```diff
@@ -243,101 +279,198 @@ const generateBoard = () => {
...
+    <button id="Ok" class="custom-button" onclick="onSelectConfirm('${suspect}')">Accept</button>
+    <button id="Cancel" class="custom-button" onclick="onSelectConfirm(false)">Cancel</button>
+    `
+  } else {
+    popupButtons.innerHTML = `
+    <button id="Ok" class="custom-button" onclick="closePopud()">Accept</button>
+     `
+  }
+
+}
```
Comment (tuttibalutti) [inline code/script.js:301]: Really nice job coming up with a custom made popup funcionality!

File: code/script.js [hunk 2]
```diff
@@ -243,101 +279,198 @@ const generateBoard = () => {
...
 // This function to start (and restart) the game
 const start = () => {
   // Here we're setting charactersInPlay array to be all the characters to start with
   charactersInPlay = CHARACTERS
   // What else should happen when we start the game?
+  generateBoard()
+  setSecret()
+  //to restart the counter
+  counter.innerHTML = ""
+  guessCounter = 0
```
Comment (tuttibalutti) [inline code/script.js:330]: Cool with a guess counter😎

File: code/script.js [hunk 3]
```diff
@@ -243,101 +279,198 @@ const generateBoard = () => {
...
-    // Similar to the one above
-  } else {
+  } 
+   else {
     if (keep) {
-      // alert popup that says something like: "Yes, the person has yellow hair! Keep all persons with yellow hair"
+      // customPopup popup that says something like: "Yes, the person has yellow hair! Keep all persons with yellow hair"
+      customPopup(
+        `Yes, the person has ${group} ${physicalAttribute}! Keep all persons with ${group} ${physicalAttribute}`, false
+      )
```
Comment (tuttibalutti) [inline code/script.js:417]: Nice string interpolation👍

File: code/script.js [hunk 4]
```diff
@@ -243,101 +279,198 @@ const generateBoard = () => {
...
+    resulText.textContent = 'Congrats! You win'
+  } else {
+    resulText.textContent = 'GAME OVER! You Lose 😖'
+  }
   // 4. Hide the game board
 }
+//Bonus Step
+const playAgain = () => {
+  start()
+  winLose.classList.remove('win-or-lose-wrapper-on')
```
Comment (tuttibalutti) [inline code/script.js:465]: Really liked that you used the classList.add/remove and the text.Content property🙌  I didn't know that the last one existed. Cool! 

I noticed that the win-or-lose-wrapper didn't cover all of the gameboard, just the viewheight (vh), so when you scroll the page the board is visible. I don't know if that is your intention or not, just a food for thought🧐

File: code/index.html [hunk 5]
```diff
@@ -59,6 +74,16 @@ <h1 id="winOrLoseText"></h1>
...
+
+      <div  id="custom-popup" class="custom-popup">
+        <div class="custom-popup-content">
+          <div id="custom-popup-text" class="custom-popup-text">
+            <h2></h2>
+          </div>
+          <div id="custom-popup-buttons" class="custom-popup-buttons">
+           </div>
+        </div>
+       </div>
```
Comment (tuttibalutti) [inline code/index.html:86]: Nice!

**Review by tuttibalutti — APPROVED** [review]
Great job Maide! I really liked that you made your own awesome Marvel-Guess Who 👏 
Super cool! Your code was clean and easy to read. Keep up the good work!

File: code/script.js (hunk 1 above)
Comment (KarolCislowski) [reply code/script.js:301]: Super nice

File: code/script.js [hunk 6]
```diff
@@ -1,231 +1,267 @@
...
+const resulText = document.getElementById('winOrLoseText')
+const popup = document.getElementById('custom-popup')
+const popupText = document.getElementById('custom-popup-text')
+const popupButtons = document.getElementById('custom-popup-buttons')
+const questions = document.getElementById('questions')
+const winLose = document.getElementById('winOrLose')
+
 
 // Array with all the characters, as objects
 const CHARACTERS = [
```
Comment (KarolCislowski) [inline code/script.js:16]: This new theme... Love it 3000 😻

File: code/script.js [hunk 7]
```diff
@@ -243,101 +279,198 @@ const generateBoard = () => {
...
   // This variable stores what option group (category) the question belongs to.
   // We also need a variable that stores the actual value of the question we've selected.
 
   if (category === 'hair color') {
     currentQuestion = {
       attribute: 'hairColor',
-      // value: ,
-      // 👆 add the value from the input here
+      value: selectedValue,
+      category: category,
```
Comment (KarolCislowski) [inline code/script.js:347]: you do not need to specify the value if you use the variable with the same name as a key, so `{category: category}` can be written as `{category}`.

**Review by KarolCislowski — APPROVED** [review]
This is a very impressive project! 🌈  You have shown great ambition 🦄 

You squeezed this juicy project to the max 🍹 🥳

**Pull request closed without merging**